app/graphql/mutations/purchase_gift_card.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Mutations
+  class PurchaseGiftCard < BaseMutation
+    argument :gift_card_id, ID, required: true
+    argument :user_id, ID, required: true
+
+    field :gift_card_order, Types::GiftCardOrderType, null: true
+
+    def execute(gift_card_id:, user_id:)
+      ensure_authorized!
+      fetch_gift_card(gift_card_id)
+      fetch_user(user_id)
+
+      # TODO: ASK should Gift card be sent rather because we need it to create it
+      result = Accrue::GiftCards::Purchase.call({user: @user, gift_card: @gift_card})
```

Comment:
```suggestion
      result = Accrue::GiftCards::Purchase.call!({user: @user, gift_card: @gift_card})
```

Using `.call!` will throw the error so you don't need to do `result.failure?` on Line 18. The GraphQL controller class should catch the `Interactor::Failure` error.